File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        insert_pos = 0
        n = len(nums)

        if n == 1:
            return n

        for i in range(n - 1):
            if nums[i] == nums[i + 1]:
                continue
            
            nums[insert_pos] = nums[i]
            insert_pos += 1
        
        nums[-1], nums[insert_pos] = nums[insert_pos], nums[-1]
        insert_pos += 1

        return insert_pos

File: 0026-remove-duplicates-from-sorted-array/0026-remove-duplicates-from-sorted-array.rs
impl Solution {
    pub fn remove_duplicates(nums: &mut Vec<i32>) -> i32 {
        let mut slow = 0;
        let n = nums.len();
        
        for fast in 1..n{
            if (nums[slow] != nums[fast]){
                slow += 1;
                nums[slow] = nums[fast]
            }
        }

        (slow + 1) as i32
    }
}